pkg/deployments/tasks/20230213-gearbox-linear-pool/test/task.fork.ts

```diff
@@ -0,0 +1,404 @@
+    MAIN_LACK,
+  }
+
+  function itRebalancesThePool(expectedState: LinearPoolState) {
+    it('rebalance the pool', async () => {
+      const { lowerTarget, upperTarget } = await pool.getTargets();
+
+      const { cash } = await vault.getPoolTokenInfo(poolId, USDC);
+      const scaledCash = cash.mul(USDC_SCALING);
+
+      let fees;
+      if (scaledCash.gt(upperTarget)) {
+        expect(expectedState).to.equal(LinearPoolState.MAIN_EXCESS);
+
+        const excess = scaledCash.sub(upperTarget);
+        fees = excess.mul(SWAP_FEE_PERCENTAGE).div(FP_ONE);
+      } else if (scaledCash.lt(lowerTarget)) {
+        expect(expectedState).to.equal(LinearPoolState.MAIN_LACK);
+
+        const lack = lowerTarget.sub(scaledCash);
+        fees = lack.mul(SWAP_FEE_PERCENTAGE).div(FP_ONE);
+      } else {
+        expect(expectedState).to.equal(LinearPoolState.BALANCED);
+
+        fees = 0;
+      }
+
+      const initialRecipientMainBalance = await mainToken.balanceOf(other.address);
+      if (expectedState != LinearPoolState.BALANCED) {
+        await rebalancer.connect(holder).rebalance(other.address);
+      } else {
+        await rebalancer.connect(holder).rebalanceWithExtraMain(other.address, 5);
+      }
+      const finalRecipientMainBalance = await mainToken.balanceOf(other.address);
+
+      if (fees > 0) {
+        // The recipient of the rebalance call should get the fees that were collected (though there's some rounding
+        // error in the main-wrapped conversion).
+        expect(finalRecipientMainBalance.sub(initialRecipientMainBalance)).to.be.almostEqual(
+          fees.div(USDC_SCALING),
+          0.0001
+        );
+      } else {
+        // The recipient of the rebalance call will get any extra main tokens that were not utilized.
+        expect(finalRecipientMainBalance).to.be.almostEqual(initialRecipientMainBalance, 0.0001);
+      }
+
+      const mainInfo = await vault.getPoolTokenInfo(poolId, USDC);
+
+      const expectedMainBalance = lowerTarget.add(upperTarget).div(2);
+      expect(mainInfo.cash.mul(USDC_SCALING)).to.equal(expectedMainBalance);
+      expect(mainInfo.managed).to.equal(0);
+    });
+  }
+
+  describe('create and check getters', () => {
+    it('deploy a linear pool', async () => {
+      const tx = await factory.create(
+        '',
+        '',
+        USDC,
+        dUSDC,
+        INITIAL_UPPER_TARGET,
+        SWAP_FEE_PERCENTAGE,
+        owner.address,
+        PROTOCOL_ID
+      );
+      const event = expectEvent.inReceipt(await tx.wait(), 'PoolCreated');
+
+      pool = await task.instanceAt('GearboxLinearPool', event.args.pool);
+      expect(await factory.isPoolFromFactory(pool.address)).to.be.true;
+
+      poolId = await pool.getPoolId();
+      const [registeredAddress] = await vault.getPool(poolId);
+      expect(registeredAddress).to.equal(pool.address);
+
+      const { assetManager } = await vault.getPoolTokenInfo(poolId, USDC); // We could query for either frxEth or mooToken
```

Comment:
```suggestion
      const { assetManager } = await vault.getPoolTokenInfo(poolId, USDC);
```

I think this is a stale comment from other fork test